zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/ZoneService.java

```diff
@@ -29,6 +113,268 @@ public ZoneResponse getZoneDetails(String zoneName) {
         return new ZoneResponse(zoneName, zone, connectionDetails);
     }
 
+    public void createZone(ZoneRequest zoneRequest) throws ZoneServiceException {
```

Comment:
This function needs refactoring as it's very difficult to read.  I think most of the code could be moved into its own functions with error handling.  Sort of like:
`public void createZone(ZoneRequest zoneRequest) throws ZoneServiceException {`
`    createSigningKey();`
`    setTokenPolicy();`
`    createDefaultIdp();`
`    createUserGroups();`
`...`
`}`

Reply:
Done